src/librustc_lexer/src/lib.rs

```diff
@@ -235,16 +236,19 @@ pub enum Base {
 /// Line won't be skipped if it represents a valid Rust syntax
 /// (e.g. "#![deny(missing_docs)]").
 pub fn strip_shebang(input: &str) -> Option<usize> {
-    debug_assert!(!input.is_empty());
-    let s: &str = &remove_whitespace(input);
-    if !s.starts_with("#!") || s.starts_with("#![") {
-        return None;
+    if let Some(first_line) = input.lines().next() {
+        let s: &str = &remove_whitespace(first_line);
+        if !s.starts_with("#!") || s.starts_with("#![") {
```

Comment:
No need to convert to strings here, `starts_with` works on `[char]` slices equally well.
`remove_whitespace` can also be inlined.

Reply:
there's no way around `collect()` ing a `Vec` instead thought, right?

Reply:
Well, you can do
```rust
        let chars = first_line.chars().filter(|c| !is_whitespace(*c));
        let three_chars = [chars.next(), chars.next(), chars.next()];
```
and then analyze characters individually, it's just inconvenient.